addons/account/migrations/9.0.1.1/post-migration.py

```diff
@@ -303,23 +402,196 @@ def account_partial_reconcile(env):
+        ON Q5.reconcile_id = amr.id
+        WHERE debit_move_id > 0 AND credit_move_id > 0
+    """)
+
+    # We want to exclude the moves that were included in the step above from
+    # the next reconciliation step.
+    cr.execute("""
+        WITH Q1 AS (
+            SELECT reconcile_id, sum(debit-credit) as balance,
+            count(id) as num_moves, count(currency_id) as num_currencies
+            FROM account_move_line
+            WHERE reconcile_id IS NOT NULL
+            GROUP BY reconcile_id, currency_id
+        ),
+        Q2 AS (
+            SELECT reconcile_id
+            FROM Q1
+            WHERE balance = 0.0
+            AND num_moves = 2
+            AND num_currencies = 0
+        )
+        SELECT aml.id
+        FROM account_move_line AS aml
+        INNER JOIN Q2
+        ON Q2.reconcile_id = aml.reconcile_id
+    """)
+
+    move_line_ids_reconciled = [move_line_id for move_line_id,
+                                in cr.fetchall()]
+
+    move_line_map = {}
+    cr.execute("SELECT reconcile_id, id "
+               "FROM account_move_line "
+               "WHERE reconcile_id IS NOT NULL "
+               "AND id NOT IN %s" % (tuple(move_line_ids_reconciled), ))
+    rec_l = {}
+    for rec_id, move_line_id in cr.fetchall():
+        move_line_map.setdefault(rec_id, []).append(move_line_id)
+        rec_l[rec_id] = True
+    num_recs = len(rec_l.keys())
+    i = 1
+    for _rec_id, move_line_ids in move_line_map.iteritems():
+        msg = 'Reconciliation step 2 (%s of %s). ' \
+              'Resolving account.move.reconcile %s.' % \
+              (i, num_recs, _rec_id)
+        openupgrade.message(cr, 'account', 'account_partial_reconcile',
+                            'id', msg)
+        move_lines = env['account.move.line'].browse(move_line_ids)
+        amount_residual_d = {}
+        for move_line in move_lines:
+            amount = abs(move_line.debit - move_line.credit)
+            amount_residual_currency = abs(move_line.amount_currency) or 0.0
+            sign = 1 if (move_line.debit - move_line.credit) > 0 else -1
+            amount_residual = move_line.company_id.currency_id.round(
+                amount * sign)
+            amount_residual_d[move_line.id] = {}
+            amount_residual_d[move_line.id]['amount_residual'] = \
+                amount_residual
+            amount_residual_d[move_line.id]['amount_residual_currency'] = \
+                move_line.currency_id and move_line.currency_id.round(
+                    amount_residual_currency * sign) or 0.0
+        auto_reconcile_lines(env, move_lines, amount_residual_d)
+        i += 1
+        move_line_ids_reconciled += move_line_ids
+
+    # The previous move lines must be flagged as reconciled. The residual
+    # amount is 0.
+    move_lines = env['account.move.line'].with_context(recompute=False).browse(
+        move_line_ids_reconciled)
+    for move_line in move_lines:
+        move_line.reconciled = True
```

Comment:
at this point, I get an unbalanced move exception. Now rerunning the migration with more logging to see what's going on there